Libraries/Cordio/controller/sources/ble/lhci/lhci_cmd_vs.c

```diff
@@ -64,6 +66,13 @@
 /**************************************************************************************************
   Functions
 **************************************************************************************************/
+//this is the callback function to the Msg
+void device_reset(wsfEventMask_t event, wsfMsgHdr_t *pMsg)
+{
+    MXC_Delay(3000);
```

Comment:
Do you need the delay?

Reply:
> Do you need the delay?

I think yes, without the delay, the reset will be completed before HCI sends the status code back.

Reply:
That delay is blocking. meaning nothing is going on when you run it.

Reply:
Instead of doing an event with a delay just use a one 
shot timer